src/types.rs

```diff
@@ -159,12 +163,58 @@ impl fmt::Display for Problem {
             write!(f, " ({})", r#type)?;
         }
 
+        if let Some(subproblems) = &self.subproblems {
+            let count = subproblems.len();
+            write!(f, ": {count} subproblems: ")?;
+            for (i, subproblem) in subproblems.iter().enumerate() {
+                write!(f, "{subproblem}")?;
+                if i != count - 1 {
+                    f.write_str(", ")?;
+                }
+            }
+        }
+
         Ok(())
     }
 }
 
 impl std::error::Error for Problem {}
 
+/// An RFC 8555 subproblem document contained within a problem returned by the ACME server
+///
+/// See <https://www.rfc-editor.org/rfc/rfc8555#section-6.7.1>
+#[derive(Clone, Debug, Deserialize)]
+pub struct Subproblem {
+    /// One of an enumerated list of problem types
+    ///
+    /// See <https://datatracker.ietf.org/doc/html/rfc8555#section-6.7>
+    pub r#type: Option<String>,
+    /// A human-readable explanation of the problem
+    pub detail: Option<String>,
+    /// The identifier associated with this problem
+    pub identifier: Option<Identifier>,
```

Comment:
Nit: maybe move the `identifier` field to be first (also matches your `Display` impl), as a sort of "key"? (Could even model it as a hashmap, but that's probably a little too opinionated.)

Reply:
Done.

> (Could even model it as a hashmap, but that's probably a little too opinionated.)

I thought about that initially but decided against it because I think the spec allows multiple subproblems for a single identifier. Making a map with `Vec` values seemed annoying so I left the type matching the spec's JSON array and figured consumers can adapt to their collection of choice.